src/net/nat/hole_punch.rs

```diff
@@ -22,7 +22,7 @@ use util;
 /// rendezvous connect to each other.
 pub fn tcp_hole_punch(
     handle: &Handle,
-    socket: TcpBuilder,
+    socket: &TcpBuilder,
```

Comment:
You can only actually "use" a `TcpBuilder` (via `listen` or `connect`) once, so it's a shame it wants to make this change. It makes more sense for this function to consume the builder.

It's too minor an issue to argue with clippy over though, so I'm happy to merge this.

Reply:
no problem, I'll revert the change and disable warning with `#[cfg_attr(feature = "cargo-clippy", allow(needless_pass_by_value))]`